python/dglke/models/ke_model.py

```diff
@@ -0,0 +1,913 @@
+from .pytorch.ke_tensor import KGEmbedding
+from .pytorch.tensor_models import cosine_dist
+from .pytorch.tensor_models import l2_dist
+from .pytorch.tensor_models import l1_dist
+from .pytorch.tensor_models import dot_dist
+from .pytorch.tensor_models import extended_jaccard_dist
+from .pytorch.tensor_models import floor_divide
+
+EMB_INIT_EPS = 2.0
+DEFAULT_INFER_BATCHSIZE = 1024
+
+class BasicGEModel(object):
+    """ Basic Graph Embeding Model
+    """
+    def __init__(self, device, model_name, score_func):
+        self._g = None
+        self._model_name = model_name
+        self._device = device
+        self._entity_emb = KGEmbedding(device)
+        self._relation_emb = KGEmbedding(device)
+        self._score_func = score_func
+
+    def attach_graph(self, g, etid_field='tid', ntid_filed='ntid'):
+        """ Attach dataset into Graph Embedding Model
+
+        Parameter
+        ----------
+        g: DGLGraph
+            Input data for knowledge graph
+        etid_field: str
+            Edge feature name storing the edge type id
+        ntid_filed: str
+            Node feature name storing the node type id
+
+        Note
+        ----
+        If the input graph is DGLGraph, we assume that it uses a homogeneous graph
+        to represent the heterogeneous graph. The edge type id is stored in etid_field
+        and the node type id is stored in ntid_filed.
+        """
+        self._etid_field = etid_field
+        self._ntid_filed = ntid_filed
+        assert isinstance(g, dgl.DGLGraph)
+        self._g = g
+
+    def load(self, model_path):
+        """ Load Graph Embedding Model from model_path.
+
+        The default entity embeding file is entity.npy.
+        The default relation embedding file is relation.npy.
+
+        Parameter
+        ---------
+        model_path : str
+            Path to store the model information
+        """
+        pass
+
+    def save(self, model_path):
+        """ Save Graph Embedding Model into model_path.
+
+        All model related data are saved under model_path.
+        The default entity embeding file is entity.npy.
+        The default relation embedding file is relation.npy.
+
+        Parameter
+        ---------
+        model_path : str
+            Path to store the model information
+        """
+        assert False, 'Not support training now'
+
+    def fit(self):
+        """ Start training
+        """
+        assert False, 'Not support training now'
+
+    def eval(self):
+        """ Start evaluation
+        """
+        assert False, 'Not support evaluation now'
+
+    def _infer_score_func(self, head, rel, tail, triplet_wise=False, batch_size=DEFAULT_INFER_BATCHSIZE):
+        head_emb = self.entity_embed[head]
+        rel_emb = self.relation_embed[rel]
+        tail_emb = self.entity_embed[tail]
+
+        num_head = head.shape[0]
+        num_rel = rel.shape[0]
+        num_tail = tail.shape[0]
+
+        score = []
+        if triplet_wise:
```

Comment:
i think you need to write comments to explain this. I'm not sure what this option means.